sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -319,6 +319,7 @@ query
 insertInto
     : INSERT OVERWRITE TABLE? multipartIdentifier (partitionSpec (IF NOT EXISTS)?)?  identifierList?        #insertOverwriteTable
     | INSERT INTO TABLE? multipartIdentifier partitionSpec? (IF NOT EXISTS)? identifierList?                #insertIntoTable
+    | INSERT INTO TABLE? multipartIdentifier REPLACE whereClause? identifierList?                           #insertIntoReplaceWhere
```

Comment:
I think the WHERE clause should be required?

Reply:
Oh, that is actually something I asked during our last chat. I think I still have some confusion. 

The grammar should be:
```
INSERT INTO TABLE? multipartIdentifier REPLACE whereClause(required) identifierList(required) 
```
as the unit test in this PR
```
INSERT INTO testcat.tbl REPLACE WHERE id = 3 SELECT * FROM source2
```

Following the notion that `? only means required` in the grammar, we should remove both `?` and the grammar becomes
```
INSERT INTO TABLE? multipartIdentifier REPLACE whereClause identifierList
```

However, this will make the unit test fail, as identifierList's `?` is removed
<img width="1548" alt="image" src="https://user-images.githubusercontent.com/114777395/201408251-6d695422-f2d6-447e-9197-cd80c1b1e905.png">

Reply:
`SELECT * FROM source2` is not identifier list, but the input query. The syntax here is for the header of INSERT and doesn't include the input query.

A full query would look like `INSERT INTO t REPLACE WHERE ... (a, c) SELECT 1, 2`.